plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/generator/TemplateProcessor.java

```diff
@@ -539,4 +570,138 @@ public AbstractConstruct caseImage(Image object) {
         }
         return super.caseImage(object);
     }
+
+    @Override
+    public AbstractConstruct caseRepresentation(Representation object) {
+        XWPFRun imageRun = insertRun(object.getStyleRun());
+        IProvider provider = ProviderRegistry.INSTANCE.getProvider(object.getRepresentationProvider());
+        if (provider == null) {
+            imageRun.setText("The image tag is referencing an unknown diagram provider.");
+            imageRun.setBold(true);
+            imageRun.setColor(ERROR_COLOR);
+        } else if (!(provider instanceof DiagramProvider)) {
+            imageRun.setText("The image tag is referencing a provider that is not an instance of DiagramProvider.");
+            imageRun.setBold(true);

```

Comment:
This is a general comment for all the TemplateProcessor : we should have a helper method that factors out the insertion of an error message in a run (Yes, I'm aware that I initially wrote this code :-)